packages/markers/src/browser/problem/problem-decorator.ts

```diff
@@ -51,17 +51,11 @@ export class ProblemDecorator implements TreeDecorator {
             }
         };
         const markers = this.appendContainerMarkers(tree, this.collectMarkers(tree));
-        processNode(tree.root);
-        const itr = new TreeNodeIterator(tree.root);
-        let node = itr.next();
-        while (!node.done) {
-            processNode(node.value);
-            node = itr.next();
-        }
+        [...new PreOrderTreeIterator(tree.root)].forEach(processNode);
```

Comment:
the same here, about inlining `processNode` with `for`

Reply:
> the same here

What same? GitHub messed it up (event without force push). I do not know what you mean. Can you give some context?

Reply:
https://github.com/theia-ide/theia/pull/1438/files/2747857805615f259dbcb89cc728bd0e7860f64a#r173074989

strangely it belongs the the review but does not appear here